### ISSUE TITLE
Flag to disable shuffling for data loader

### ISSUE THREAD

**Issue by hasansalimkanmaz** [huggingface/transformers#11693]
# 🚀 Feature request

Currently, Trainer is shuffling the train_dataset by default and there is no flag to enable/disable it. 

@sgugger  

## Motivation

Even if shuffling the dataset brings a lot of benefits like preventing overfitting, at some point, one can need to disable it for experimental motivation. It isn't possible to do it without overwriting the `_get_train_sampler` method of Trainer. :( 

## Your contribution

I can work on this issue (maybe next month) if this issue gets positive feedback.

**Comment by sgugger** [comment]
I don't think this is a suitable feature, so I would recommend you override and subclass `get_train_dataloader` to simply return a training dataloader without using a sampler or `shuffle=True`.
